internal/agent/device/config/controller.go

```diff
@@ -14,15 +14,9 @@ import (
 	"github.com/flightctl/flightctl/pkg/log"
 )
 
-type Controller interface {
-	Initialize(ctx context.Context, current *v1alpha1.RenderedDeviceSpec)
-	Sync(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error
-	WriteIgnitionFiles(ctx context.Context, files []ignv3types.File) error
```

Comment:
note: exposing this meant the bootstrap controller had access to write ign. this was removed to ensure the only writes of files are the spec manager and config controller during Sync.